data-prepper-plugins/otel-metrics-raw-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/otelmetrics/OTelMetricsRawProcessor.java

```diff
@@ -85,7 +81,7 @@ public Collection<Record<? extends Metric>> doExecute(Collection<Record<?>> reco
         for (Record<?> rec : records) {
             if ((rec.getData() instanceof Event)) {
                 Record<? extends Metric> newRecord = (Record<? extends Metric>)rec;
-                if (otelMetricsRawProcessorConfig.getFlattenAttributesFlag() ||
+                if (!otelMetricsRawProcessorConfig.getFlattenAttributesFlag() ||
```

Comment:
Can you explain why this is being negated?

Reply:
It's true by default. Only when it is set to false in the config, we call `JacksonMetric.setFlattenAttributes(false)`